catalog/transformer/catalog-transformer-xlsx/src/main/java/ddf/catalog/transformer/xlsx/XlsxMetacardUtility.java

```diff
@@ -0,0 +1,118 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.xlsx;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.BinaryContentImpl;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.Font;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XlsxMetacardUtility {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(XlsxMetacardUtility.class);
+
+  private static MimeType mimeType = new MimeType();
+
+  static {
+    try {
+      mimeType.setPrimaryType("application");
+      mimeType.setSubType("vnd.openxmlformats-officedocument.spreadsheetml.sheet");
+    } catch (MimeTypeParseException e) {
+      LOGGER.warn("Failure creating MIME type", e);
+      throw new ExceptionInInitializerError(e);
+    }
+  }
+
+  public BinaryContent buildSpreadSheet(List<Metacard> metacards) {
+
+    Set<AttributeDescriptor> attributeDescriptors =
+        metacards.get(0).getMetacardType().getAttributeDescriptors();
+
+    int rowIndex = 0;
+    int cellIndex = 0;
+
+    Workbook workbook = new XSSFWorkbook();
+    Sheet sheet = workbook.createSheet();
+    Row row = sheet.createRow(rowIndex++);
+
+    CellStyle style = workbook.createCellStyle();
+    Font font = workbook.createFont();
+    font.setBold(true);
+    style.setFont(font);
+
+    // Write header row.
+    for (AttributeDescriptor attributeDescriptor : attributeDescriptors) {
+      String attributeName = attributeDescriptor.getName();
+      Cell cell = row.createCell(cellIndex++);
+      cell.setCellValue(attributeName);
+      cell.setCellStyle(style);
+    }
+
+    for (Metacard metacard : metacards) {
+      List<String> values = getMetacardValues(attributeDescriptors, metacard);
+      row = sheet.createRow(rowIndex++);
+
+      cellIndex = 0;
+      for (String value : values) {
+        row.createCell(cellIndex++).setCellValue(value);
+      }
+    }
+
+    return writeWorkbook(workbook);
+  }
+
+  private BinaryContent writeWorkbook(Workbook workbook) {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
```

Comment:
❗️ Ensure that we close this output stream using a try-catch with resources otherwise we'll have a memory leak.

Reply:
Looks like ByteArrayOutputStream doesn't need closing
https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html#close()

Reply:
Imo, even though `ByteArrayOutputStream` doesn't need to be closed, it is good practice to always close streams, which will prevent devs from having to know which streams need to be closed and which streams do not need to be closed.